pip install --upgrade tiktoken

In [39]:
import tiktoken

print(f'I tokenizzatori / embedder disponibili sono:\n{tiktoken.list_encoding_names()}')
encoder_id = 'o200k_base'
tokenizer = tiktoken.get_encoding(encoder_id)
# in alternativa si può specificare il nome del modello:
#tokenizer = tiktoken.encoding_for_model('gpt-4o')

print(f"\nIl vocabulario di '{encoder_id}' ha {tokenizer.n_vocab} token")

I tokenizzatori / embedder disponibili sono:
['gpt2', 'r50k_base', 'p50k_base', 'p50k_edit', 'cl100k_base', 'o200k_base']

Il vocabulario di 'o200k_base' ha 200019 token


| Il vocabulario di | è | e ha |
|---|---|---|
| gpt-2 | r50k_base | 50257 token |
| gpt-3.5 e gpt-4 | cl100k_base | 100277 token |
| gpt-4o | o200k_base | 200019 token |

In [51]:
tokenizer.encode_single_token('good')

30744

In [54]:
text = "La bellezza salverà il mondo."

print(f'Il testo originale: {text}')

tokens = tokenizer.encode(text)
print(f'Gli id dei token: {tokens}')
print(f'I token: {[token.decode('utf-8') for token in tokenizer.decode_tokens_bytes(tokens)]}')

decoded = tokenizer.decode(tokens)
print(f'I token decodificati: {decoded}')

Il testo originale: La bellezza salverà il mondo.
Gli id dei token: [4579, 1912, 168053, 2370, 445, 708, 1793, 57990, 13]
I token: ['La', ' bel', 'lezza', ' sal', 'ver', 'à', ' il', ' mondo', '.']
I token decodificati: La bellezza salverà il mondo.


Nota che tiktoken non contiene funzioni per calcolare i vettori di embedding dei token: per questo occorre un modello.